Sources/swift-format/Run.swift

```diff
@@ -17,6 +17,42 @@ import SwiftFormatConfiguration
 import SwiftFormatCore
 import SwiftSyntax
 
+/// Runs the linting pipeline over a source file that will be read from standard input. Optionally,
+/// a filename can be provided to use in the output diagnostics.
+///
+/// If there were any lint diagnostics emitted, this function returns a non-zero exit code.
+/// - Parameters:
+///   - configuration: The `Configuration` that contains user-specific settings.
+///   - assumingFilename: The filename of the file that contains the source code.
+/// - Returns: Zero if there were no lint errors, otherwise a non-zero number.
+func lintMain(configuration: Configuration, assumingFilename: String?) -> Int {
```

Comment:
I'm a bit uncomfortable having separate entry points for the `stdin` case and the file case because any logic that we may want to modify in one will need to be kept in sync with the other.

The main tricky thing here is that the `stdin` entry point calls the API that takes a source string and assumed URL, and the file-based API takes the URL directly.

What if we have this function take a `FileHandle` and an assumed filename, and then this function can always read from the file handle and call the `(source:assumingFileURL:)` API? Then the caller of this function can pass in either `FileHandle.standardInput` or a `FileHandle` to the requested file on disk, and we can converge on a single implementation of `lintMain` and `formatMain`.

That means that the `swift-format` tool will no longer use the versions of the API that take a URL directly, but that's fine—those APIs are still useful to have for other potential tool integrations aside from ours.

Reply:
That makes sense. I agree having to keep changes in sync is not ideal. I've refactored the existing methods (deleting the new methods) to use FileHandle arguments to read the source code.

I ended up refactoring main.swift a bit to share the code in the FileIterator loop, since I needed to add some more identical code to that loop.